### PR TITLE
add some tests and github action

### PR DESCRIPTION
**Pull request by wolfv**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        run: |
+          export MAMBA_ROOT_PREFIX=~/micromamba
+          export MAMBA_EXE=$(pwd)/micromamba
+          . $MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh
+          ./micromamba create -y -p ~/test_env 'python>=3.7' fastapi typer authlib \
+                       httpx=0.12.0 sqlalchemy sqlite python-multipart uvicorn zstandard \
+                       conda-build appdirs toml fsspec pytest -c conda-forge
+        env:
+          PYTHON_VERSION: ${{ matrix.python-version }}
+      - name: build tests
```
Comment (adriendelsalle) [inline .github/workflows/ci.yml:49]: run test instead of build tests?

File: quetz/config.py [hunk 2]
```diff
@@ -79,7 +84,9 @@ class Config:
     _config_dirs = [_site_dir, _user_dir]
     _config_files = [os.path.join(d, _filename) for d in _config_dirs]
 
-    def __init__(self, deployment_config: str = None) -> NoReturn:
+    def __init__(
+        self, deployment_config_file: str = None, config: Dict = {}, db_session=None
```
Comment (adriendelsalle) [inline quetz/config.py:88]: add arg `config` as a convenience for testing ?

File: quetz/config.py [hunk 3]
```diff
@@ -88,23 +95,30 @@ def __init__(self, deployment_config: str = None) -> NoReturn:
 
         Parameters
         ----------
-        deployment_config : str, optional
+        deployment_config_file : str, optional
             The configuration stored at deployment level
+        config : Dict, optional
+            The configuration to apply (overrides deployment_config_file)
```
Comment (adriendelsalle) [inline quetz/config.py:101]: missing `db_session` docstring

File: quetz/config.py (hunk 2 above)
Comment (adriendelsalle) [inline quetz/config.py:88]: missinfg type hint for `db_session`

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          mkdir -p ~/micromamba/pkgs/
+      - name: install deps
+        shell: bash -l {0}
+        run: |
+          export MAMBA_ROOT_PREFIX=~/micromamba
+          export MAMBA_EXE=$(pwd)/micromamba
+          . $MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh
+          ./micromamba create -y -p ~/test_env 'python>=3.7' fastapi typer authlib \
+                       httpx=0.12.0 sqlalchemy sqlite python-multipart uvicorn zstandard \
+                       conda-build appdirs toml fsspec pytest -c conda-forge
```
Comment (adriendelsalle) [inline .github/workflows/ci.yml:46]: `conda-build` not required

File: quetz/main.py [hunk 5]
```diff
@@ -37,36 +37,46 @@
...
 from quetz.dao import Dao
-from quetz.database import get_session as get_db_session
 
 from .condainfo import CondaInfo
 
 app = FastAPI()
 
-config = Config()
-auth_github.register(config)
+if "QUETZ_IS_TEST" in os.environ:
```
Comment (adriendelsalle) [inline quetz/main.py:45]: used for testing with pytest?
why not creating an `app` fixture with this configuration instead?

I can't see the `QUETZ_IS_TEST` env var in the CI. Are the tests passing?

File: quetz/tests/test_db_models.py [hunk 6]
```diff
@@ -18,3 +21,50 @@ def test_user(db):
...
+    assert len(response.text)
+    response = client.get('/api/channels')
+
+    response = client.get('/api/channels', headers={"X-Api-Key": keya})
+    assert len(response.json()) == 2
+    response = client.get('/api/channels', headers={"X-Api-Key": keyb})
+    assert len(response.json()) == 1
+
+
+def test_api(db):
```
Comment (adriendelsalle) [inline quetz/tests/test_db_models.py:63]: usefull method vs `test_api_key`?

**Review by adriendelsalle — COMMENTED** [review]
Hi @wolfv !
If I may add few comments on this PR, I hope this will help.
Thank you for this!

**Comment by wolfv** [comment]
Hi @adriendelsalle thanks for the comments. I agree with all of them :)

File: quetz/main.py (hunk 5 above)
Comment (wolfv) [reply quetz/main.py:45]: The problem is that (right now) we need the app object in this file to attach all routes to it.
However, when we create routers, we can attach them to the app object and refactor, which will be much nicer (and allow us to create the fixture). I'll take care of that today.

File: quetz/tests/test_db_models.py (hunk 6 above)
Comment (wolfv) [reply quetz/tests/test_db_models.py:63]: nope :)

**Pull request merged**